### DISCUSSION TITLE
[Help]: Cannot find module 'embla-carousel/components/Engine' or its corresponding type declarations.ts(2307)

### DISCUSSION THREAD

**Discussion by imeite** [davidjerleke/embla-carousel#732] [Help] [answered]
### Summary

Hi , I'm a newbie programmer and using Embla Carousel pre-defined example: Infinite Scroll.
I copied most the applicable source code directly from the sandbox (REACT+TYPESCRIPT).
In brief, it's working fine in my project.

Issue:
import { EngineType } from 'embla-carousel/components/Engine'

It gives this typescript error:
Cannot find module 'embla-carousel/components/Engine' or its corresponding type declarations.ts(2307)

The path 'embla-carousel/components/Engine' doesn't seem to exist. 
Does anyone know the correct path to use?

### If applicable, which variants of Embla Carousel are relevant to this question?

- [ ] embla-carousel **(Core)**
- [X] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

### Additional information

```js
import { EngineType } from 'embla-carousel/components/Engine'

It gives this typescript error:
Cannot find module 'embla-carousel/components/Engine' or its corresponding type declarations.ts(2307)

Package.json:
{
  "name": "carousel",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "@types/node": "20.11.13",
    "@types/react": "18.2.48",
    "@types/react-dom": "18.2.18",
    "embla-carousel-react": "^8.0.0-rc22",
    "eslint": "8.56.0",
    "eslint-config-next": "14.1.0",
    "next": "14.1.0",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "typescript": "5.3.3"
  }
}
```

**Comment by davidjerleke** [comment, marked as answer]
Hi @imeite.

Thanks for your question. Could your problem be related to this perhaps?

- https://github.com/davidjerleke/embla-carousel/discussions/726#discussioncomment-8331335

Best,
David

**Reply by imeite** [reply]
Changing "moduleResolution" from "bundle" to "node" fixed it.
Sorry i did searched online and here to make sure there wasn't anything already tickets out there and didn't find anything. 
I think i searched in the "bug" sections only.
Thank you!

**Comment by davidjerleke** [comment]
- Duplicate of #726.
